ingestors/kafka/src/main/scala/hydra/kafka/util/ConsumerGroupsOffsetConsumer.scala

```diff
@@ -0,0 +1,223 @@
+import hydra.kafka.algebras.{KafkaAdminAlgebra, KafkaClientAlgebra}
+import hydra.kafka.model.TopicConsumer.{TopicConsumerKey, TopicConsumerValue}
+import hydra.kafka.model.TopicConsumerOffset.{TopicConsumerOffsetKey, TopicConsumerOffsetValue}
+import hydra.kafka.model.TopicMetadataV2Request.Subject
+import hydra.kafka.model.{TopicConsumer, TopicConsumerOffset}
+import io.chrisdavenport.log4cats.Logger
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
+import io.confluent.kafka.serializers.{AbstractKafkaAvroSerDeConfig, KafkaAvroSerializer}
+import kafka.common.OffsetAndMetadata
+import kafka.coordinator.group.{BaseKey, GroupMetadataManager, OffsetKey}
+import org.apache.avro.generic.GenericRecord
+import org.apache.kafka.common.TopicPartition
+
+import scala.collection.JavaConverters._
+import scala.collection.SortedSet
+import scala.util.Try
+
+object ConsumerGroupsOffsetConsumer {
+
+  var myMap = (0 to 50).map((_, 0L)).toMap
+
+  def start[F[_]: ContextShift: ConcurrentEffect: Timer: Logger](
+                                                                  kafkaClientAlgebra: KafkaClientAlgebra[F],
+                                                                  kafkaAdminAlgebra: KafkaAdminAlgebra[F],
+                                                                  schemaRegistryAlgebra: SchemaRegistry[F],
+                                                                  uniquePerNodeConsumerGroup: String,
+                                                                  consumerOffsetsOffsetsTopicConfig: Subject,
+                                                                  kafkaInternalTopic: String,
+                                                                  dvsConsumersTopic: Subject,
+                                                                  bootstrapServers: String,
+                                                                  commonConsumerGroup: String
+                                                                ): F[Unit] = {
+    val dvsConsumerOffsetStream = kafkaClientAlgebra.consumeMessagesWithOffsetInfo(consumerOffsetsOffsetsTopicConfig.value, uniquePerNodeConsumerGroup, commitOffsets = false)
+
+    for {
+      schemaRegistryClient <- schemaRegistryAlgebra.getSchemaRegistryClient
+      deferred <- Deferred[F, PartitionOffsetMap]
+      hydraConsumerOffsetsOffsetsLatestOffsets <- kafkaAdminAlgebra.getLatestOffsets(consumerOffsetsOffsetsTopicConfig.value).map(_.map(l => l._1.partition -> l._2.value))
+      hydraConsumerOffsetsOffsetsCache <- Ref[F].of[PartitionOffsetMap](hydraConsumerOffsetsOffsetsLatestOffsets.filter(_._2 == 0))
+      initialPartitionCache <- initializePartitions(kafkaAdminAlgebra, kafkaInternalTopic)
+      backgroundProcess <- Concurrent[F].start(getOffsetsToSeekTo(initialPartitionCache, deferred, dvsConsumerOffsetStream,hydraConsumerOffsetsOffsetsLatestOffsets, hydraConsumerOffsetsOffsetsCache))
+      partitionMap <- deferred.get
```

Comment:
The deferred in an earlier version of this was causing a delay on startup. Is this still the case?

Reply:
This entire class is being called from a background thread so it will halt execution on that thread, but not the whole process.